### DISCUSSION TITLE
Question about memory usage

### DISCUSSION THREAD

**Discussion by Tsuna77** [jellyfin/jellyfin#12191] [Q&A]
Hello,

I'm using jellyfin in a docker image (jellyfin/jellyfin:latest) with ressource limitation (10 CPU and 20GB of RAM)

I see that jellyfin process used all memory available. In this screen you can see memory and streams statistics (get by a curl every 30s and store in a pg databases)

![image](https://github.com/jellyfin/jellyfin/assets/3943654/46adda55-af08-4f2d-a89d-b435d95d692a)

Why jellyfin do not free this memory ? I recently upgrade my jellyfin hardware, it run without any problem with less than 10GB and now it's using 20G in only 36 hours of life :(

Am I the only one with that ?

**Comment by cvium** [comment]
There are many reports but the issue is not reproducible by us. Which plugins do you have installed?

**Reply by Tsuna77** [reply]
I used those plugins :

- AudioDB 10.9.7.0
- Fanart 11.0.0.0
- MusicBrainz 10.9.7.0
- OMDb 19.9.7.0
- Open Subtitles 20.0.0.0
- Playback Reporting 14.0.0.0
- Reports 17.0.0.0
- Session Cleaner 3.0.0.0
- SkinManager 1.5.0.0
- Studio Images 10.9.7.0
- TMDb 10.9.7.0
- TheTVDB 13.0.0.0
- Theme Songs 10.9.2.0

**Reply by cvium** [reply]
Can you try without playback reporting and session cleaner?

**Reply by Tsuna77** [reply]
uninstalled, I restart jellyfin. I'm also using jellystat and if I remember it's need to have Playback reporting to get stats ^^

I'll see tomorrow if memory is not too high

**Reply by Tsuna77** [reply]
I restarted it again this evening and in less than 2 hours, is was going up to 12.5Gb 

![image](https://github.com/jellyfin/jellyfin/assets/3943654/b97967d1-86a6-4845-8168-76e2f2b2e523)

**Reply by cvium** [reply]
Am I reading this right? You have 6-9 streams active during this time of high memory usage? 1-2 of those are transcodes, which will always consume memory. Does the memory usage drop once the streams are done?

**Reply by Tsuna77** [reply]
you are reading right, and yes there is a drop 2 hours after those 7 stream, transcode stream are not full transcoding but multiplexing

the next graph start at the same time as the last one but finish now

![image](https://github.com/jellyfin/jellyfin/assets/3943654/2f076990-a4f9-4cd8-83f4-08a7be7ee4b7)

Max was 16.8Gb and at 2am drop to 12Gb and get to 13.5Gb now without all thoose streams

**Reply by Tsuna77** [reply]
I just had to restart jellyfin, because it goes to 20Gb and do not respond anymore to some api endpoint like sessions (used for grafana)

as you say memory usage increase a lot when sending video and or without transcode but do not reduce after :(

![image](https://github.com/jellyfin/jellyfin/assets/3943654/4d980929-7241-4afe-bee3-28dcff84b966)


This time I had 4 stream including 1 transcode (so 3 direct play)

**Reply by Tsuna77** [reply]
I found a problem in my metrics, ans maybe it's why you can not reproduce. I was using container_memory_usage_bytes{image!=""} in grafana, but this one also include cache memory. If I switch to container_memory_working_set_bytes{image!=""} I go from 20Gb to 7Gb that is better.

Maybe the problem is in memory kept in cache to long ?

**Reply by Tsuna77** [reply]
you can see both memory graph here :
![image](https://github.com/jellyfin/jellyfin/assets/3943654/f8017de1-2c7a-4ed6-99ca-24d63f416074)

**Reply by cvium** [reply]
Can you please try setting this env var in the container? `DOTNET_GCHighMemPercent = 48`
